src/client/react/Components/Admin/AdminEventRow.jsx

```diff
@@ -20,6 +27,41 @@ export default function AdminEventRow(props) {
     hasChanged.current = true;
   }, [event]);
 
+  const accessToken = window.boUserData.access_token
+  const apiUrl = window.boClientConfig.baseUrl
+
+  const getUsers = async () => {
+    if (!loading) {
+      setIsLoading(true);
+
+      await axios.get(`${apiUrl}/event/${event.id}/participants/`, { headers: { authorization: `Bearer ${accessToken}` } })
+        .then((userListJson) => {
+          setListOfUsers(userListJson.data);
+          setIsLoading(false);
+        }).catch((e) => {
+          setIsLoading(false);
+          console.log(e);
+        });
+      csvLink.current.link.click();
```

Comment:
Ist dieses `link.click` über einen React-ref irgendwo in der Bibliothek dokumentiert? Sieht mir aus als könnte das bei einem Update der Dependency jederzeit kaputt gehen, was wir vermeiden sollten.

Damit zusammenhängend: in der [Dokumentation von CSV-Link](https://github.com/react-csv/react-csv) ist die `onClick`-Prop dokumentiert, die eine asynchrone Methode aufruft und dort kann dann das Laden des CSV-Inhalts geschehen. Dort ist sogar ein Beispiel mit axios direkt in der Readme.

Ich würde vorschlagen auch den dokumentierten Weg zu gehen und nicht den über Refs und undokumentierte Methoden.

Reply:
Ist jetzt wie dokumentiert mit der CsvLink-Component umgesetzt.